prime-router/src/main/kotlin/tokens/Scope.kt

```diff
@@ -72,19 +114,93 @@ class Scope {
             val splits = scope.split(".")
             val orgName = splits[0]
             val senderOrReceiver = splits[1]
-            val detailedScope = DetailedScope.valueOf(splits[2])
+            val detailedScope: DetailedScope = DetailedScope.toDetailedScope(splits[2])!! // checked in isValidScope
             return Triple(orgName, senderOrReceiver, detailedScope)
         }
 
         /**
```

Comment:
All calls to authorize anything in reportstream should end up calling `authorizedScopes` here, which is intentionally very dumb - it makes no attempt know anything about the meaning of the scope strings.

If there is any intersection between the authenticated scopes the user claims, and the scopes required to use a resource, then the user is authorized.